keras/src/saving/saving_lib.py

```diff
@@ -1080,3 +1095,10 @@ def get_attr_skiplist(obj_type):
         f"saving_attr_skiplist_{obj_type}", skiplist
     )
     return skiplist
+
+
+def is_large_model(model):
+    if weight_memory_size(model.variables) > _LARGE_MODEL_THRESHOLD:
```

Comment:
Ideally this could depend on available RAM -- unfortunately that seems relatively difficult to achieve in a cross-platform setting (we could use `psutil`, maybe).

Reply:
I have utilized `psutil` to determine whether to write/read the model weigths into memory based on available system memory.
The default upper bound is set to 50%, which is an arbitrary number I picked.
Please let me know if we can replace it with a more reasonable number.